src/node/routes/index.ts

```diff
@@ -170,7 +164,7 @@ export const register = async (
 
   const wsErrorHandler: express.ErrorRequestHandler = async (err, req, res, next) => {
     logger.error(`${err.message} ${err.stack}`)
-    ;(req as WebsocketRequest).ws.end()
+    ;(req as pluginapi.WebsocketRequest).ws.end()
```

Comment:
question: why do we have to assert this? (previous work, but maybe we could add a comment explaining why)

Reply:
The `ErrorRequestHandler` type marks `req` as `Request` so we have to cast it as a websocket request.

But really what we should do is create our own `WsErrorRequestHandler` type so we don't have to do this.

Reply:
Creating a custom error handler type is turning out to be weirdly difficult so I'm going to leave this for now.